docs/extend/modular-resources/upload/_index.md

```diff
@@ -0,0 +1,211 @@
+      You can later increment this value with subsequent `viam module upload` commands.
+   - `platform` - provide one of the following, depending on the platform you have built your custom module for (You can use the `uname -m` command to determine your system architecture):
+      - `darwin/arm64` - macOS computers running the `arm64` architecture, such as Apple Silicon.
+      - `darwin/amd64` - macOS computers running the Intel `x86_64` architecture.
+      - `linux/arm64` - Linux computers or {{< glossary_tooltip term_id="board" text="boards" >}} running the `arm64` (`aarch64`) architecture, such as the Raspberry Pi.
+      - `linux/amd64` - Linux computers or {{< glossary_tooltip term_id="board" text="boards" >}} running the Intel `x86_64` architecture.
+   - `path` - provide the path to the compressed archive, in `tar.gz` or `tar.xz` format, that contains your custom module code.
+
+   For example, the following command uploads the compressed `my-module.tar.gz` archive to the Viam Registry when run in the same directory as the corresponding `meta.json` file:
+
+   ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   viam module upload --version 1.0.0 --platform darwin/arm64 my-module.tar.gz
+   ```
+
+For more information, see the [`viam module` command](/manage/cli/#module)
+
+## Update an existing module
+
+You can also use the [Viam CLI](/manage/cli/) to update an existing custom module in the Viam Registry.
+
+1. Edit your custom module with the changes you'd like to make.
+
+1. Update your custom module's `meta.json` file with the changes, if any.
+   For example, if you have altered your model's name, or adjusted the endpoint name, you'll need to update `meta.json` with these changes.
+
+1. Run `viam module update` to register the configuration changes to your module (and to `meta.json` if applicable):
+
+   - To register a *public* module, run the following command from within the same directory as your `meta.json` file:
+
+      ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+      viam module update
+      ```
+
+   - To register a *private* module, run the following command from within the same directory as your `meta.json` file, providing your organization ID:
+
+      ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+      viam module update --org-id <org-id>
+      ```
+
+   On successful update, the command will return a link to the updated module in the Viam Registry.
+
+1. Re-package your custom module to get it ready to upload to the Viam Registry. Currently, the Registry only supports `tar.gz` or `tar.xz` format. For example, run the following command to compress your custom module as a `tar.gz` archive:
+
+   ``` sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+   tar -zcf <packaged-module.tar.gz> </path/to/module-directory>
```

Comment:
I suggest
```
touch module.tar.gz
tar -czf module.tar.gz --exclude=module.tar.gz .
```
This will help prevent `tar: .: file changed as we read it` errors (doesn't seem to happen on macOS but happens on linux)

Reply:
Landed on explicit choosing of include files/dirs with:

```
touch module.tar.gz
tar -czf module.tar.gz --exclude=module.tar.gz run.sh [requirements.txt] src
```